codegenerator/cli/templates/static/codegen/src/eventFetching/FetchState.res

```diff
@@ -843,9 +857,13 @@ Check the max queue size of the tip of the tree.
 Don't use the cummulative queue sizes because otherwise there
 could be a deadlock. With a very small buffer size of the actively
 fetching registration
+ 
+If there are pending dynamic contracts, we always need to allow the next query
 */
-let isReadyForNextQuery = (self: t, ~maxQueueSize) =>
-  self.baseRegister.fetchedEventQueue->Array.length < maxQueueSize
+let isReadyForNextQuery = ({pendingDynamicContracts, baseRegister}: t, ~maxQueueSize) =>
+  pendingDynamicContracts->Utils.Array.isEmpty
+    ? baseRegister.fetchedEventQueue->Array.length < maxQueueSize
+    : true
```

Comment:
Won't be the case when there are always some `pendingDynamicContracts`?

Reply:
There shouldn't always be pending dynamic contracts, whenever a query gets made or returns the pending dynamic contracts turn into registers

Reply:
But they might be in the next query and so on. While we are already exceeded the queue size. Well it's 100% better than getting stack.

Reply:
Yes but we have the same problem with registers.  Fortunately they try and merge as soon as they can.

I've said many times I wish I never named the variable "MAX_QUEUE_SIZE" because it's really a threshold and not a hard limit.